### PR TITLE
Explicitly mention BM25 in score docs

### PR DESCRIPTION
**Pull request by romseygeek**

### CODE REVIEW

File: docs/general/ddl/system-columns.rst [hunk 1]
```diff
@@ -56,11 +56,15 @@ and might contain underscores in between.
 
 ``_score``
   This internal system column is available on all documents retrieved by a
-  ``SELECT`` query. It is representing the scoring ratio of the document
-  related to the used query filter and makes most sense on fulltext searches.
-
-  The scoring ratio is always related to the highest score determined by a
-  search, thus scores are not directly comparable across searches.
+  ``SELECT`` query. It represents the score of the document as returned by
+  the query filter, and makes most sense for full text queries.
```
Comment (amotl) [inline docs/general/ddl/system-columns.rst:60]: While being at it, I just wanted to add guidance about the "why so?" details. I hope you get the idea, and will be able to come up with something more fluent.
```suggestion
  the query filter, and makes most sense for full text queries, by using it
  as a rank indicator to know how good the expression matches the
  document, compared to others.
```

**Review by amotl — CHANGES_REQUESTED** [review]
Thank you Alan. I've just added some suggestions. Feel free to use them or not.

File: docs/general/ddl/system-columns.rst [hunk 2]
```diff
@@ -56,11 +56,15 @@ and might contain underscores in between.
...
+
+  The score is calculated using Lucene's BM25 Similarity implementation, which
```
[the remark this answers is not in the document]
Comment (romseygeek) [reply docs/general/ddl/system-columns.rst:62]: I tried to find a decent page to link to here, but there is surprisingly little on the public web.  There's the [lucene javadoc](https://lucene.apache.org/core/9_9_0/core/org/apache/lucene/search/similarities/BM25Similarity.html) but that's pretty sparse, and then there's a scattering of blog posts from elastic and open source connections, but they're fairly specific to elasticsearch or Solr.  There's also the [Wikipedia page for Okapi BM25](https://en.wikipedia.org/wiki/Okapi_BM25) which gives a good overview of the algorithm, but our implementation is subtly different.

Maybe I should link to both the lucene javadoc and the Wikipedia page, and say something like "The score is calculated using Lucene's BM25Similarity implementation, based on the widely-used Okapi BM25 algorithm"?

File: docs/general/ddl/system-columns.rst [hunk 3]
```diff
@@ -56,15 +56,26 @@ and might contain underscores in between.
...
-
-  The scoring ratio is always related to the highest score determined by a
-  search, thus scores are not directly comparable across searches.
+  ``SELECT`` query. It represents the score of the document as returned by
+  the query filter, and makes most sense for full text queries, giving an
+  indication of how well the query expression matches the document.
+  Documents with higher scores match the query more closely, allowing the
+  score to be used for ranking.
+
+  The score is calculated using Lucene's `BM25Similarity implementation`,
```
Comment (hlcianfagna) [inline docs/general/ddl/system-columns.rst:65]:
```suggestion
  The score is calculated using Lucene's `BM25Similarity implementation`_ ,
```

File: docs/general/ddl/system-columns.rst [hunk 4]
```diff
@@ -56,15 +56,26 @@ and might contain underscores in between.
...
+  based on the widely-used `Okapi BM25` algorithm.  This uses a combination
```
Comment (hlcianfagna) [inline docs/general/ddl/system-columns.rst:66]:
```suggestion
  based on the widely-used `Okapi BM25`_ algorithm.  This uses a combination
```

**Review by hlcianfagna — APPROVED** [review]

File: docs/general/ddl/system-columns.rst [hunk 5]
```diff
@@ -56,15 +56,26 @@ and might contain underscores in between.
...
-  ``SELECT`` query. It is representing the scoring ratio of the document
-  related to the used query filter and makes most sense on fulltext searches.
-
-  The scoring ratio is always related to the highest score determined by a
-  search, thus scores are not directly comparable across searches.
+  ``SELECT`` query. It represents the score of the document as returned by
+  the query filter, and makes most sense for full text queries, giving an
+  indication of how well the query expression matches the document.
+  Documents with higher scores match the query more closely, allowing the
+  score to be used for ranking.
```
Comment (amotl) [inline docs/general/ddl/system-columns.rst:63]: Excellent! Thanks.

File: docs/general/ddl/system-columns.rst [hunk 6]
```diff
@@ -56,15 +56,26 @@ and might contain underscores in between.
...
+
+  The score is calculated using Lucene's `BM25Similarity implementation`_ ,
+  based on the widely-used `Okapi BM25`_ algorithm.  This uses a combination
```
Comment (amotl) [inline docs/general/ddl/system-columns.rst:66]: Perfect. Thanks.

**Review by amotl — APPROVED** [review]

**Comment by ckurze** [comment]
@romseygeek @amotl I just had a look at https://cratedb.com/docs/crate/reference/en/master/general/ddl/system-columns.html - the links are unfortunately not working. Could you pls. check?

Thanks

**Comment by romseygeek** [comment]
Weird, I thought we'd fixed those.  I've opened https://github.com/crate/crate/pull/15396

**Pull request merged**